forward/request_sender.go

```diff
@@ -145,18 +152,38 @@ func (s *requestSender) MakeCall(ctx context.Context, res *[]byte) <-chan error
 		if s.format == tchannel.Thrift {
 			_, arg3, _, err = raw.WriteArgs(call, []byte{0, 0}, s.request)
 		} else {
-			_, arg3, _, err = raw.WriteArgs(call, nil, s.request)
+			var resp *tchannel.OutboundCallResponse
+			_, arg3, resp, err = raw.WriteArgs(call, nil, s.request)
+
+			// check if the response is an application level error
+			if err == nil && resp.ApplicationError() {
+				// parse the json from the application level error
+				errResp := struct {
```

Comment:
Not to my knowledge, they create the struct inline as well where it gets send. I can make a type in this file if it makes it more readable.

https://github.com/uber/tchannel-go/blob/master/json/handler.go#L156-L162